src/v/raft/log_eviction_stm.cc

```diff
@@ -16,23 +16,161 @@
 
 namespace raft {
 
+ss::sstring
+log_eviction_stm::eviction_event_type_to_str(eviction_event_type type) {
+    switch (type) {
+    case eviction_event_type::storage:
+        return "storage";
+    case eviction_event_type::kafka:
+        return "kafka";
+    }
+    return "undefined";
+}
+
 log_eviction_stm::log_eviction_stm(
   consensus* raft,
   ss::logger& logger,
   ss::lw_shared_ptr<storage::stm_manager> stm_manager,
   ss::abort_source& as)
-  : _raft(raft)
+  : raft::state_machine(raft, logger, ss::default_priority_class())
   , _logger(logger)
   , _stm_manager(std::move(stm_manager))
   , _as(as) {}
 
 ss::future<> log_eviction_stm::start() {
     monitor_log_eviction();
     process_log_eviction();
-    return ss::now();
+    set_next(model::next_offset(_raft->flushed_offset()));
+    return raft::state_machine::start();
+}
+
+static bool is_within_bounds(
+  ss::logger& logger,
+  ss::lw_shared_ptr<const storage::offset_translator_state> translator,
+  model::offset kafka_truncation_offset,
```

Comment:
I think we should do the translation somewhere in the caller layer all raft operates on log offsets, this introduces a notion of something Kafka specific into that world. Maybe perform translation in `kafka::replicated_partition` ?